packages/deployment/test/test_kes.js

```diff
@@ -0,0 +1,167 @@
+  t.is(result, resourceString);
+});
+
+test('lookupLambdaReference throws error on invalid configuration', (t) => {
+  const resourceString = '${SomeOtherConfigurationString.foobar}';
+  const kes = setupKesForLookupLambdaReference(t.context.kes, 'FunctionName', null);
+  const error = t.throws(() => {
+    kes.lookupLambdaReference(resourceString);
+  }, Error);
+  t.is(error.message, `Invalid stateObjectResource: ${resourceString}`);
+});
+
+test('injectWorkflowLambdaAliases updates the correct resources', (t) => {
+  const kes = t.context.kes;
+
+  kes.config.lambdas = {
+    TestLambda: { hash: 'notarealhash' },
+    TestUnversionedLambda: { hash: null }
+  };
+
+  kes.config.stepFunctions = {
+    TestStepFunction: {
+      States: {
+        1: { Type: 'Task', Resource: '${TestLambdaLambdaFunction.Arn}' },
+        2: { Type: 'Task', Resource: '${TestUnversionedLambdaLambdaFunction.Arn}' },
+        3: { Type: 'Task', Resource: '${SomethingElse.Arn}'}
+      }
+    },
+    TestStepFunction2: {
+      States: {
+        1: { Type: 'Task', Resource: '${TestLambdaLambdaFunction.Arn}' }
+      }
+    }
+  };
+
+  const expected = {
+    TestStepFunction: {
+      States: {
+        1: { Type: 'Task', Resource: '${TestLambdaLambdaAliasnotarealhash}' },
+        2: { Type: 'Task', Resource: '${TestUnversionedLambdaLambdaFunction.Arn}' },
+        3: { Type: 'Task', Resource: '${SomethingElse.Arn}'}
+      }
+    },
+    TestStepFunction2: {
+      States: {
+        1: { Type: 'Task', Resource: '${TestLambdaLambdaAliasnotarealhash}' }
+      }
+    }
+  };
+
+  kes.injectWorkflowLambdaAliases();
+  t.deepEqual(expected, kes.config.stepFunctions);
+});
+
+
+test('injectOldWorkflowLambdaAliases adds oldLambdas to configuration object', async (t) => {
+  const kes = t.context.kes;
+  const nameArray = ['LambdaName-12345', 'LambdaName-abcde', 'SecondLambdaName-67890'];
+  kes.getRetainedLambdaAliasNames = () => (Promise.resolve(nameArray));
+
+  const expected = {
+    LambdaName: { hashes: ['12345', 'abcde'] },
+    SecondLambdaName: { hashes: ['67890'] }
+  };
+
+  await kes.injectOldWorkflowLambdaAliases();
+  t.deepEqual(expected, kes.config.oldLambdas);
+});
+
+
+test('parseAliasName parses the alias name', (t) => {
+  const expected = { name: 'functionName', hash: 'hashValue' };
+  const actual = t.context.kes.parseAliasName('functionName-hashValue');
+  t.deepEqual(expected, actual);
+});
+
+test('parseAliasName returns null hash if hash missing', (t) => {
+  const expected = { name: 'functionName', hash: null };
+  const actual = t.context.kes.parseAliasName('functionName-');
+  t.deepEqual(expected, actual);
+});
+
+
+test.serial('getAllLambdaAliases returns an unpaginated list of aliases', async (t) => {
+  const kes = t.context.kes;
+
+  // Mock out AWS calls
+  const versionUpAliases = aliasFixture.Aliases[0];
+
+  const listAliasesStub = sinon.stub().callsFake(() => Promise.reject());
```

Comment:
Now we're just getting into things I don't understand.  Why do you set this calls fake to rejected promises and then with onCall below have it return the paginated fakes?

Reply:
That sets the *default* return response for the stub.   Then in lines 133-139 we set the response for calls 0->versionUpAliases.length-1     That way if anything in the tested code calls past the end of the fixture the stub rejects the promise and the test fails.